bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventInteractor.kt

```diff
@@ -0,0 +1,135 @@
+package io.bucketeer.sdk.android.internal.event
+
+import androidx.annotation.VisibleForTesting
+import io.bucketeer.sdk.android.BKTException
+import io.bucketeer.sdk.android.internal.Clock
+import io.bucketeer.sdk.android.internal.IdGenerator
+import io.bucketeer.sdk.android.internal.event.db.EventDao
+import io.bucketeer.sdk.android.internal.logd
+import io.bucketeer.sdk.android.internal.model.Evaluation
+import io.bucketeer.sdk.android.internal.model.Event
+import io.bucketeer.sdk.android.internal.model.User
+import io.bucketeer.sdk.android.internal.remote.ApiClient
+import io.bucketeer.sdk.android.internal.remote.RegisterEventsResult
+import java.util.concurrent.atomic.AtomicReference
+
+internal class EventInteractor(
+  private val eventsMaxBatchQueueCount: Int,
+  private val apiClient: ApiClient,
+  private val eventDao: EventDao,
+  private val clock: Clock,
+  private val idGenerator: IdGenerator,
+) {
+
+  @VisibleForTesting
+  internal val events: AtomicReference<List<Event>> = AtomicReference(emptyList())
+
+  private var eventUpdateListener: EventUpdateListener? = null
+
+  fun setEventUpdateListener(listener: EventUpdateListener?) {
```

Comment:
Just confirming, who will listen to event updates?

Reply:
https://gist.github.com/yshrsmz/3690b1085dfe7edd1ec2218d9880bb38

EventForegroundScheduler will use this listener to send events when the cache exceeds the threshold and reschedule itself.